regressiontests/regression_test.go

```diff
@@ -0,0 +1,435 @@
+	"github.com/bitcoin-sv/block-headers-service/internal/chaincfg"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/headers"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/merkleroots"
+	"github.com/bitcoin-sv/block-headers-service/transports/http/endpoints/api/tips"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const localHTTPServerURL = "http://localhost:8080"
+const whatsonchainAPIURL = "https://api.whatsonchain.com/v1/bsv/main/chain/tips"
+
+type WhatsOnChainForkTip struct {
+	Height    int    `json:"height"`
+	Hash      string `json:"hash"`
+	BranchLen int    `json:"branchlen"`
+	Status    string `json:"status"`
+}
+
+var dbEngine string
+
+func init() {
+	flag.StringVar(&dbEngine, "dbEngine", "postgres", "The database engine to use in tests (postgres or sqlite)")
+}
+
+func TestApplicationIntegration(t *testing.T) {
+	flag.Parse()
+	ctx := context.Background()
+
+	setupApplicationBuild(t)
+	defer cleanupApplicationExecutable(t)
+
+	var cmd *exec.Cmd
+	var dbCleanupFunc func()
+
+	switch dbEngine {
+	case "postgres":
+		cmd, dbCleanupFunc = setupPostgresDatabase(ctx, t)
+	case "sqlite":
+		cmd, dbCleanupFunc = setupSQLiteDatabase(t)
+	default:
+		t.Fatalf("Unsupported database engine: %s", dbEngine)
+	}
+
+	defer dbCleanupFunc()
+	defer killApplicationProcess(cmd, t)
+
+	monitorApplicationStartup(ctx, t, cmd)
+
+	t.Log("Comparing synced data with known checkpoints...")
+
+	verifyHeaders(ctx, t)
+
+	t.Log("Headers checkpoint comparison passed successfully 🎉")
+
+	t.Log("Verifying merkle roots...")
+
+	verifyMerkleRoots(ctx, t, fixtures)
+
+	t.Log("Merkle roots verification passed successfully 🎉")
+
+}
+
+func setupApplicationBuild(t *testing.T) {
+	buildCmd := exec.Command("go", "build", "-o", pathToExecutable(), "../cmd/main.go")
+	if err := buildCmd.Run(); err != nil {
+		t.Fatalf("Failed to build the application for tests: %v", err)
+	}
+}
+
+func cleanupApplicationExecutable(t *testing.T) {
+	if err := os.Remove(pathToExecutable()); err != nil {
+		t.Logf("Couldn't remove executable files: %v", err)
+	}
+}
+
+func setupPostgresDatabase(ctx context.Context, t *testing.T) (*exec.Cmd, func()) {
+	postgresContainer, mappedPort := startPostgresContainer(ctx, t)
+	cleanupFunc := func() {
+		if err := postgresContainer.Terminate(ctx); err != nil {
+			t.Logf("Couldn't terminate postgres container: %v", err)
+		}
+	}
+	cmd := configureApplication("postgres", mappedPort)
+	return cmd, cleanupFunc
+}
+
+// setupSQLiteDatabase sets up the environment for SQLite and returns the command to start the application.
+func setupSQLiteDatabase(t *testing.T) (*exec.Cmd, func()) {
+	cmd := configureApplication("sqlite", "")
+	cleanupFunc := func() {
+		cleanupSQLiteDB(t)
+	}
+	return cmd, cleanupFunc
+}
+
+func configureApplication(engine, postgresPort string) *exec.Cmd {
+	cmd := exec.Command(pathToExecutable())
+	cmd.Env = os.Environ()
```

Comment:
Without using current environment variables the `os.TempDir()` used in `database/import.go/getHeadersFile` returned `C:/windows` directory as a temp dir. The problem was that the program has not permissions for that directory. 

Need to be re-tested whether my modification doesn't break something on mac/linux.